File: example/src/main/java/io/castle/android/sample/MainActivity.java
/*
 * Copyright (c) 2017 Castle
 */

package io.castle.android.sample;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.widget.Button;

import java.util.HashMap;
import java.util.Map;

import butterknife.ButterKnife;
import butterknife.OnClick;
import io.castle.android.Castle;

public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.main);

        ButterKnife.bind(this);
    }

    @OnClick(R.id.identify)
    public void onIdentifyClick(Button button) {
        // Set signature for secure mode
        Castle.secure("944d7d6c5187cafac297785bbf6de0136a2e10f31788e92b2822f5cfd407fa52");

        // Identify user with a unique identifier including user traits
        Map<String, String> traits = new HashMap<>();
        traits.put("email", "sebastian@boldsie.com");
        Castle.identify("sebastiansimson", traits);
    }

    @OnClick(R.id.track_screen)
    public void onTrackScreenClick(Button button) {
        // Track a screen view and include some properties
        Map<String, String> properties = new HashMap<>();
        properties.put("deviceOrientation", "horizontal");
        Castle.screen("Menu", properties);
    }

    @OnClick(R.id.flush)
    public void onFlushClick(Button button) {
        Castle.flush();
    }

    @OnClick(R.id.reset)
    public void onResetClick(Button button) {
        Castle.reset();
    }
}

File: example/src/main/java/io/castle/android/sample/MainActivity.kt
/*
 * Copyright (c) 2020 Castle
 */
package io.castle.android.sample

import android.Manifest
import android.content.pm.PackageManager
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.core.app.ActivityCompat
import androidx.core.content.ContextCompat
import io.castle.android.Castle
import io.castle.android.sample.databinding.MainBinding
import java.util.*

class MainActivity : AppCompatActivity() {
    private val LOCATION_PERMISSION_REQUESST_CODE: Int = 1001
    private lateinit var binding: MainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = MainBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)

        binding.userJwt.setOnClickListener { onUserJwtClick() }
        binding.trackScreen.setOnClickListener { onTrackScreenClick() }
        binding.trackCustom.setOnClickListener { onTrackCustomClick() }
        binding.flush.setOnClickListener { onFlushClick() }
        binding.reset.setOnClickListener { onResetClick() }
        binding.location.setOnClickListener { onLocationClick() }
    }

    private fun onLocationClick() {
        requestLocationAccess()
    }

    private fun requestLocationAccess() {
        if (ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
            // Permission is not granted
            ActivityCompat.requestPermissions(this, arrayOf(Manifest.permission.ACCESS_FINE_LOCATION), LOCATION_PERMISSION_REQUESST_CODE)
        } else {
            Toast.makeText(this,"Permission already granted", Toast.LENGTH_SHORT).show()
        }
    }

    private fun onUserJwtClick() {
        // Identify with user encoded as jwt
        Castle.userJwt("eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6ImVjMjQ0ZjMwLTM0MzItNGJiYy04OGYxLTFlM2ZjMDFiYzFmZSIsImVtYWlsIjoidGVzdEBleGFtcGxlLmNvbSIsInJlZ2lzdGVyZWRfYXQiOiIyMDIyLTAxLTAxVDA5OjA2OjE0LjgwM1oifQ.eAwehcXZDBBrJClaE0bkO9XAr4U3vqKUpyZ-d3SxnH0")
    }

    private fun onTrackScreenClick() {
        // Track a screen view
        Castle.screen("Menu")
    }

    private fun onTrackCustomClick() {
        Castle.custom("Added to cart", mapOf(
            "product" to "iPhone 13 Pro",
            "price" to 1099.99
        ))
    }

    private fun onFlushClick() {
        Castle.flush()
    }

    private fun onResetClick() {
        Castle.reset()
    }
}